src/web/app/pages-student/student-home-page/student-home-page.component.ts

```diff
@@ -64,63 +54,82 @@ export class StudentHomePageComponent implements OnInit {
 
   user: string = '';
 
-  recentlyJoinedCourseId?: string = '';
-  hasEventualConsistencyMsg: boolean = false;
   courses: StudentCourse[] = [];
-  sessionsInfoMap: { [key: string]: SessionInfoMap } = {};
 
   constructor(private route: ActivatedRoute, private httpRequestService: HttpRequestService,
-              private statusMessageService: StatusMessageService, private loadingBarService: LoadingBarService) { }
+              private statusMessageService: StatusMessageService,
+              private feedbackSessionsService: FeedbackSessionsService,
+              private timezoneService: TimezoneService,
+              private loadingBarService: LoadingBarService) {
+    this.timezoneService.getTzVersion();
+  }
 
   ngOnInit(): void {
     this.route.queryParams.subscribe((queryParams: any) => {
       this.user = queryParams.user;
-      this.getStudentCourses(queryParams.persistencecourse);
+      this.getStudentCourses();
     });
   }
 
   /**
    * Gets the courses and feedback sessions involving the student.
    */
-  getStudentCourses(persistencecourse: string): void {
+  getStudentCourses(): void {
     this.loadingBarService.showLoadingBar();
-    const paramMap: { [key: string]: string } = { persistencecourse };
-    this.httpRequestService.get('/student/courses', paramMap)
-        .pipe(finalize(() => this.loadingBarService.hideLoadingBar()))
-        .subscribe((resp: StudentCourses) => {
-          this.recentlyJoinedCourseId = resp.recentlyJoinedCourseId;
-          this.hasEventualConsistencyMsg = resp.hasEventualConsistencyMsg;
-          this.courses = resp.courses;
-          this.sessionsInfoMap = resp.sessionsInfoMap;
-
-          if (this.hasEventualConsistencyMsg) {
-            this.statusMessageService.showWarningMessage(
-                'You have successfully joined the course ' + `${this.recentlyJoinedCourseId}` + '. '
-                + 'Updating of the course data on our servers is currently in progress '
-                + 'and will be completed in a few minutes. '
-                + 'Please refresh this page in a few minutes to see the course ' + `${this.recentlyJoinedCourseId}`
-                + ' in the list below.');
-          }
-
-        }, (resp: ErrorMessageOutput) => {
-          this.statusMessageService.showErrorMessage(resp.error.message);
-        });
+    const paramMap: { [key: string]: string } = {
+      entitytype: 'student',
+    };
+    this.httpRequestService.get('/courses', paramMap).subscribe((resp: Courses) => {
+      for (const course of resp.courses) {
+        this.feedbackSessionsService.getFeedbackSessionsForStudent(course.courseId)
+            .pipe(finalize(() => this.loadingBarService.hideLoadingBar()))
```

Comment:
There is an inconsistency between the time you call `showLoadingBar` and `hideLoadingBar `.